pyqtorch/gpsr.py

```diff
@@ -100,7 +100,7 @@ def forward(
         ctx.embedding = embedding
         values = param_dict(param_names, param_values)
         ctx.out_state = circuit.run(state, values)
-        ctx.projected_state = observable.run(ctx.out_state, values)
+        ctx.projected_state = observable(ctx.out_state, values)
```

Comment:
Currently only the `QuantumCircuit` has a `run()` method. All other "composite" operators follow the same interface as individual operations: the `forward` method returns the statevector after applying the operator, $|\psi\rangle = \hat{O}|\phi\rangle$. For individual operators, that means simply applying the operator itself. For composite operators, that means applying each of the "leaf" operators according to some rule: the `Sequence` applies each one in sequence, the `Add` applies each one separately and sums the outcome, the `Scale` applies the leaf operator and multiplies with the scale. The `Observable` is just a wrapper on top of `Add`, so it will behave exactly the same, with the added convenience of there being an `expectation` method that returns $\langle\phi|\hat{O}|\phi\rangle$.